docs/design/proposals/direct-blk-device-assignment.md

```diff
@@ -0,0 +1,264 @@
+* **NodeStageVolume/NodeUnStageVolume** -- Invoke a kata-runtime handler to clean up the stale mount information.
+Typical block device based CSI plugins (like EBS, Azure, GCE PD, Ceph, etc) 
+mounts the PV to a global mount path in the host during NodeStageVolume, which can be skipped in this case.
+
+### Kata-runtime changes
+The host agent (CSI node driver, as an example) will call into kata-runtime. Kata-runtime behaves as the API for 
+sandbox-level management interactions. In order for the CSI node driver to communicate with the kata-runtime, 
+we propose augmenting kata-runtime to include a filesystem, `fs`, command as follows:
+
+```
+kata-runtime fs —help
+NAME:
+kata-runtime fs [subcommand] [options]— interact with mounted filesystem volume
+
+SUBCOMMANDS:
+ - add-blk-mount-info <mountinfo> -- add the mount info of a volume to the Kata runtime.
+ - reset-blk-mount-info <host-blk-device-name> –- clean up the mount info directory under kata runtime
+ - get-stats <host-blk-device-name> — show status and usage details for the mounted filesystem backed by the given block device, <host-blk-device-name>.
+ - resize <host-blk-device-name> <size> — update the mounted volume to <size> for the mounted filesystem backed by the given block device, <host-blk-device-name>.
+
+```
+The mountinfo object is defined as follows:
+```Golang
+type MountInfo struct {
+    // Device: source device path for this volume
+    Device string json:"device,omitempty"
+    // FsType: filesystem that needs to be used to mount the storage inside the VM
+    FsType string json:"fsType,omitempty"
+    // metadata field for easy extension
+    Metadata map[string]string json:"metadata,omitempty"
+    // Options: additional options that might be needed to mount the storage filesystem
+    Options []string json:"options,omitempty"
+}
+```
+The volume stas is defined as below following the [csi spec](https://github.com/container-storage-interface/spec/blob/master/lib/go/csi/csi.pb.go).
+```Golang
+type NodeGetVolumeStats struct {
+	// This field is OPTIONAL.
+	Usage []*VolumeUsage `json:"usage,omitempty"`
+	// Information about the current condition of the volume.
+	// This field is OPTIONAL.
+	// This field MUST be specified if the VOLUME_CONDITION node
+	// capability is supported.
+	VolumeCondition      *VolumeCondition `json:"volume_condition,omitempty"`
+}
+```
+
+#### Quick notes on implementation
+
+Instead of the CSI node driver writing the mount info into a `csiPlugin.json` file under the volume root, 
+as described in the original proposal. Here we propose that the CSI node driver passes the mount information to 
+the kata-runtime through a new kata-runtime command called `add-blk-mount-info`. The kata-runtime then write the mount 
+information to a `“`mount-info.json`”` file in a hardcoded location (e.g. `/run/kata-containers/shared/direct-block-devices/[block_dev_name]/`).
```

Comment:
Please don't use "unicode quotes".

Reply:
As @jodh-intel said, there are some full-width double/single quotes, you'd better replace the whole doc.

@jodh-intel maybe feng is using a Chinese input soft(IME), to switch between English/Chinese, usually, one touch of SHIFT key is used, to input a double quote, SHIFT is needed too, therefore, it is easy to mix Chinese characters when entering English.